entrypoint.sh

```diff
@@ -20,35 +15,47 @@ then
   exit 1
 fi
 
-if [ -z "$GITHUB_HEAD_REF" ]
+if [ -z "$GITHUB_RUN_ID" ]
 then
-  # we have a push event
-  BRANCH=${GITHUB_REF:11}
-else
-  BRANCH=${GITHUB_HEAD_REF}
+  echo "Must specify GITHUB_RUN_ID"
+  exit 1
 fi
 
-# jq queries
-jq_run_ids=".workflow_runs | .[] | select(.head_branch==\"${BRANCH}\" and (.status==\"in_progress\" or .status==\"queued\" or .status==\"waiting\")) | .id"
-
-# get the github workflow ID
-
 GITHUB_API=https://api.github.com
 
 auth_header="Authorization: token ${GITHUB_TOKEN}"
 
-workflow_url=$(curl -s ${GITHUB_API}/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID} -H "${auth_header}" | jq -r '.workflow_url' | cut -d/ -f2-)
-workflow_id=${workflow_url##/*/}
-echo "workflow id: "$workflow_id
+function extractMetaInformation() {
+  jq "{ workflow_id: .workflow_id, branch: .head_branch, repo: .head_repository.full_name}"
+}
+
+function convertToKeyValuePairs() {
+  jq -r "to_entries | map(\"\(.key)=\(.value | tostring)\") | .[]"
+}
+
+function getRunningWorkflowIds() {
+  jq ".workflow_runs | .[] | select(.head_branch==\"${branch}\" and .head_repository.full_name==\"${repo}\" and .status==\"in_progress\" or .status==\"queued\" or .status== \"waiting\") | .id "
+}
+
+function exportAll() {
+  for var in $1; do
+    export $var
+  done
+}
+
+# extract meta information for current workflow run
+exportAll "$(curl -s "${GITHUB_API}/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}" -H "${auth_header}" | extractMetaInformation | convertToKeyValuePairs)"
```

Comment:
should we set the `"Accept: application/vnd.github.v3+json"` header here aswell? 
This [documentation](https://docs.github.com/en/rest/reference/actions#get-a-workflow-run) recommends that.

Reply:
I didn't see your reaction in time.. fixed.